src/mito2/src/region/opener.rs

```diff
@@ -476,11 +474,8 @@ pub(crate) async fn replay_memtable<S: LogStore>(
         region_write_ctx.write_memtable();
     }
 
-    if allow_stale_entries && stale_entry_found {
-        wal.obsolete(region_id, flushed_entry_id, wal_options)
-            .await?;
-        info!("Force obsolete WAL entries, region id: {}, flushed entry id: {}, last entry id read: {}", region_id, flushed_entry_id, last_entry_id);
-    }
+    wal.obsolete(region_id, flushed_entry_id, wal_options)
```

Comment:
Maybe we can only do this when the last_id of the log store is not equal to `flushed_entry_id`.

Reply:
> Maybe we can only do this when the last_id of the log store is not equal to `flushed_entry_id`.

`LogStore` does not have a `span` API that returns current min/max entry id (only raft-engine provide that functionality).

Reply:
I think we should keep the info log.

```rust
info!("Force obsolete WAL entries, region id: {}, flushed entry id: {}, last entry id read: {}", region_id, flushed_entry_id, last_entry_id);
```